src/components/shooting/ShootingView.tsx

```diff
@@ -0,0 +1,273 @@
+      if (rendererRef.current && cameraRef.current) {
+        const w = window.innerWidth;
+        const h = window.innerHeight;
+        rendererRef.current.setSize(w, h);
+        cameraRef.current.aspect = w / h;
+        cameraRef.current.updateProjectionMatrix();
+      }
+    };
+
+    window.addEventListener('resize', handleResize);
+
+    const tick = () => {
+      if (tweenGroupRef.current) {
+        // ふわふわの処理
+        tweenGroupRef.current.update();
+      }
+      if (controlsRef.current) {
+        controlsRef.current.update();
+      }
+      if (rendererRef.current && sceneRef.current && cameraRef.current) {
+        rendererRef.current.render(sceneRef.current, cameraRef.current);
+      }
+
+      requestAnimationFrame(tick);
+    };
+
+    tick();
+
+    return () => {
+      window.removeEventListener('resize', handleResize);
+      if (rendererRef.current) {
+        elm?.removeChild(rendererRef.current.domElement);
+      }
+    };
+  }, [avatars]);
+
+  const handlePointerDown = (event: React.PointerEvent<HTMLDivElement>) => {
+    if (cameraRef.current && raycasterRef.current && sceneRef.current) {
+      const x = (event.clientX / window.innerWidth) * 2 - 1;
+      const y = -(event.clientY / window.innerHeight) * 2 + 1;
+
+      raycasterRef.current.setFromCamera(
+        new Three.Vector2(x, y),
+        cameraRef.current,
+      );
+
+      // calculate objects intersecting the picking ray
+      const intersects = raycasterRef.current.intersectObjects(
+        sceneRef.current.children,
+      );
+      let count = 0;
+      for (const obj of intersects) {
+        if (obj.object.userData.avatar) {
+          const avatar = obj.object.userData.avatar as Avatar3DObject;
+          avatar.object.removeFromParent();
+          setStock((prev) => [...prev, avatar]);
+          if (stock.length + count === 4) {
+            setDialogOpen(true);
+          }
+          count++;
+        }
+      }
+    }
+  };
+
+  const goToNext = () => {};
+
+  return (
+    <div>
+      <Video
+        disable={!orientationSupported}
+        className={!orientationSupported ? 'hidden' : ''}
+        style={{ position: 'fixed', left: 0, top: 0 }}
+      />
+      <div
+        ref={mountRef}
+        onPointerDown={handlePointerDown}
+        className='bg-transparent fixed top-0 left-0'
+        style={{ position: 'fixed', left: 0, top: 0 }}
+      />
+      <StockView avatars={stock} />
+
+      <Dialog open={dialogOpen} onOpenChange={() => goToNext()}>
+        <DialogContent>
+          <DialogHeader>
+            <DialogTitle>スタックがいっぱいになりました。</DialogTitle>
+            <DialogDescription>選別に進みますか？</DialogDescription>
+          </DialogHeader>
+          <DialogFooter>
+            <DialogClose asChild>
+              <Button type='button' variant='secondary'>
+                はい
+              </Button>
+            </DialogClose>
+          </DialogFooter>
+        </DialogContent>
+      </Dialog>
+    </div>
```

Comment:
https://ja.react.dev/reference/react/Fragment
```suggestion
    <>
      <Video
        disable={!orientationSupported}
        className={!orientationSupported ? 'hidden' : ''}
        style={{ position: 'fixed', left: 0, top: 0 }}
      />
      <div
        ref={mountRef}
        onPointerDown={handlePointerDown}
        className='bg-transparent fixed top-0 left-0'
        style={{ position: 'fixed', left: 0, top: 0 }}
      />
      <StockView avatars={stock} />

      <Dialog open={dialogOpen} onOpenChange={() => goToNext()}>
        <DialogContent>
          <DialogHeader>
            <DialogTitle>スタックがいっぱいになりました。</DialogTitle>
            <DialogDescription>選別に進みますか？</DialogDescription>
          </DialogHeader>
          <DialogFooter>
            <DialogClose asChild>
              <Button type='button' variant='secondary'>
                はい
              </Button>
            </DialogClose>
          </DialogFooter>
        </DialogContent>
      </Dialog>
    </>
```